src/libs/actions/User.js

```diff
@@ -83,17 +86,31 @@ function updatePassword(oldPassword, password) {
  * @param {String} message optional reason for closing account
  */
 function closeAccount(message) {
-    DeprecatedAPI.User_Delete({message}).then((response) => {
-        console.debug('User_Delete: ', JSON.stringify(response));
-
-        if (response.jsonCode === 200) {
-            Growl.show(Localize.translateLocal('closeAccountPage.closeAccountSuccess'), CONST.GROWL.SUCCESS);
-            redirectToSignIn();
-            return;
-        }
-
-        // Inform user that they are currently unable to close their account
-        CloseAccountActions.showCloseAccountModal();
+    API.write('CloseAccount', {message}, {
+        optimisticData: [
+            {
+                onyxMethod: 'merge',
+                key: ONYXKEYS.CLOSE_ACCOUNT,
+                value: {isLoading: true},
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: 'clear',
+            },
+            {
+                onyxMethod: 'merge',
+                key: ONYXKEYS.NVP_PREFERRED_LOCALE,
+                value: preferredLocale,
+            },
+        ],
```

Comment:
The server is returning the `successData` response already? Should we also do it here?

Reply:
All that being said, are there any concerns you have with this?

Reply:
> Technically the only way that Pusher can send data is if "the server" tells it to do so 🤺

😆 "a server", yes! "the server" though? 😉 What really is "the" server? hahahaha

> If we are going to use successData it's for loading states and other things.

Was this in a doc somewhere that I missed? Looking around at what has been merged so far, I don't see `successData` used much **except** one instance of setting `loading: false` so I see what you mean - I have been thinking about other ways `successData` is useful and I could think of clearing errors, not much else 😅 

About the Pusher points - I also hope Pusher is "rock solid" / doesn't need any backup mechanisms / failsafes - this will make code cleaner too which would be nice. I'm convinced - no need to set the `successData` here since it's a duplicate of the Pusher data we'll receive, which should be enough 👍

Reply:
Just to confirm we're on the same page, it's also not even needed to clear the `isLoading` state in `successData` in this case b/c the Pusher updates will clear all Onyx data, right? 

If we don't clear `isLoading` in `successData`, I think it would be useful to add a comment explaining why not (since it's not immediately apparent that our server will send a Pusher update to clear Onyx)

Reply:
Everything you are saying here makes sense and sounds good to me 👍 

>Was this in a doc somewhere that I missed? Looking around at what has been merged so far, I don't see successData used much except one instance of setting loading: false so I see what you mean - I have been thinking about other ways successData is useful and I could think of clearing errors, not much else

I'm not sure if this made it into a doc explicitly, but the closest thing I found is [this Slack discussion](https://expensify.slack.com/archives/C02HWMSMZEC/p1652977595063489?thread_ts=1652899459.602989&cid=C02HWMSMZEC).